.github/workflows/test-ui.yml

```diff
@@ -47,12 +47,29 @@ jobs:
         run: cd ui && yarn run lint:js
       - name: lint:hbs
         run: cd ui && yarn run lint:hbs
-      - name: Ember tests
+  tests:
+    needs:
+      - lint
+    strategy:
+      matrix:
+        partition: [1, 2, 3, 4]
+        split: [4]
+    uses: ./.github/workflows/_ember-exam.yml
+    with:
+      partition: ${{ matrix.partition }}
+      split: ${{ matrix.split }}
+  finalize:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    needs:
+      - tests
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
+      - name: yarn install
+        run: cd ui && yarn install --frozen-lockfile
+      - name: finalize
         env:
           PERCY_TOKEN: ${{ secrets.PERCY_TOKEN }}
-        run: mkdir -p /tmp/test-reports && cd ui && yarn exam
-      - uses: actions/upload-artifact@0b7f8abb1508181956e8e162db84b466c27e18ce # v3.1.2
-        with:
-          path: "/tmp/test-reports"
+        run: cd ui && yarn percy -- build:finalize
```

Comment:
glancing at [the percy docs](https://docs.percy.io/docs/cli-build#percy-buildfinalize), I'm not sure I understand what `finalize` does exactly, and how is it different from `wait`?  I assume it brings the parallel bits back together somehow.

Reply:
It works like `wait` except it coordinates across machines/containers. It also integrates natively with GHA so we don't see any of the IDing and synchronizing of partitions that it's doing behind the scenes.

Relevant docs: https://docs.percy.io/docs/parallel-test-suites#parallel-suites-split-across-different-machines

Reply:
oh neat.  a bit lower down the page there though, it indicates that GHA needs some extra hand-holding, because `GITHUB_RUN_ID` is not unique across reruns?

https://docs.percy.io/docs/parallel-test-suites#advanced-nonce-setting

the example they give though is a ~deprecated way to set GHA step outputs, so you'd want something like this instead:

```
run: echo "result=${{ github.run_id }}-$(date +%s)" >> "$GITHUB_OUTPUT"
```

ref: https://docs.github.com/en/actions/using-jobs/defining-outputs-for-jobs